green_cli/common.py

```diff
@@ -346,12 +346,13 @@ def getreceiveaddress(session, details):
 
 @green.command()
 @click.option('--subaccount', default=0, expose_value=False, callback=details_json)
-@click.option('--last-pointer', default=0, expose_value=False, callback=details_json)
+@click.option('--last-pointer', default=None, expose_value=False, callback=details_json)
 @with_login
 @print_result
 @with_gdk_resolve
 def getpreviousaddresses(session, details):
     """Get previously generated addresses."""
+    details['last_pointer'] = details.get('last_pointer')
```

Comment:
The way this is implemented in gdk at the moment means it's optional, so we could skip this line, although not sure if the optionality of the field is officially supported?

Reply:
gdk expected the null value to be set and behaved incorrectly if the field was missing.
I changed that on the gdk side so that newest addresses are obtained if "last_pointer" is null or the key is missing, then I removed the above line